ddev/ddev
# fix: escape composer args, for #6604

## The Issue

From Discord https://discord.com/channels/664580571770388500/1296719909367054420

```
ddev composer config repositories.0 "{\"type\": \"composer\", \"url\":\"hello\",  \"exclude\": [\"magento/composer-dependency-version-audit-plugin\"]}"
```

(We can pass a json)

Since today (1.23.5), it fails with this message: 

```
In ConfigCommand.php line 766:You must pass the type and a url. Example: php composer.phar config reposit  ories.foo vcs https://bar.com
```
As I'm passing a type and url, I suspect there are lost somewhere

The culprit:
- #6604

This was discussed earlier in:
- #2547

## How This PR Solves The Issue

Adds proper escaping.
Adds missing test coverage.

## Manual Testing Instructions

Test any composer command, that has `"`, `'`, `' '` (whitespace), or `{}`:

```
ddev composer config description "test description"
ddev composer config description "''''"
ddev composer config description "\""
ddev composer config repositories.local '{"type": "composer", "url": "https://example.com"}'
```

And check `composer.json` file for changes.

## Automated Testing Overview



## Release/Deployment Notes
